test-common/flatpak-spawn.c

```diff
@@ -304,7 +313,8 @@ flatpak_populate_runtime (GFile        *updater_dir,
 
   g_autoptr(GFile) repo_directory_path = g_file_new_for_path (repo_directory);
 
-  g_key_file_set_string (metadata, "Runtime", "name", runtime_name);
+  g_key_file_set_string (metadata, "Runtime", "name", name);
+  g_key_file_set_string (metadata, "Runtime", "runtime", runtime_name);
```

Comment:
This change looks suspicious. It’s from commit e2184f70c0611cd292b87bc60487fd6d48e5ee0a, but there’s no explanation of why it’s necessary.

Reply:
Yeah I should had added a review comment explaining it. Its there because a flatpak `Runtime` entry needs a `runtime` specifying the fully-qualified runtime ref name, eg:

```
smspillaz@endless:~/Source/eos-updater$ cat /var/lib/flatpak/runtime/com.endlessm.Platform/x86_64/master/active/metadata 
[Runtime]
name=com.endlessm.Platform/x86_64/master
runtime=com.endlessm.Platform/x86_64/master
sdk=com.endlessm.Sdk/x86_64/master
```

Reply:
Yeah, weird, it crashes if I use `runtime_name` in place of "name":

```
./tests/test-update-install-flatpaks 
/updater/install-no-flatpaks: OK
/updater/install-flatpaks-pull-to-repo: **
eos-updater-test-common:ERROR:test-common/convenience.c:200:etc_update_client: 'cmd_result_ensure_all_ok_verbose (cmds)' should be TRUE
Aborted (core dumped)
cleaning up pid 100323
```

I suspect that may be because we read the runtime metadata somewhere?

Reply:
I don’t think we read the runtime metadata within the eos-updater code. libflatpak might well do. Could you investigate further? At the least it looks to me like `name` and `runtime_name` need some comment (and renaming?) to distinguish them in the source code, since one is fully-qualified and the other isn’t.

Reply:
Created https://phabricator.endlessm.com/T20880 as a follow-up